src/wp-admin/post.php

```diff
@@ -239,26 +239,26 @@
 		check_admin_referer( 'trash-post_' . $post_id );
 
 		if ( ! $post ) {
-			wp_die( __( 'The item you are trying to move to the Trash no longer exists.' ) );
+			wp_die( __( 'The item you are trying to move to the Trash no longer exists.' ), 404 );
```

Comment:
Maybe `410 Gone` would be more appropriate? https://www.rfc-editor.org/rfc/rfc9110#section-15.5.11